src/main/java/fi/fmi/avi/model/swx/immutable/SpaceWeatherAdvisoryImpl.java

```diff
@@ -105,6 +118,62 @@ public Builder addAllPhenomena(final List<SpaceWeatherPhenomenon> elements) {
             return super.addAllPhenomena(elements);
         }
 
+        public SpaceWeatherAdvisoryImpl.Builder withCompleteIssueTimeNear(final ZonedDateTime reference) {
+            requireNonNull(reference, "reference");
+            return mapIssueTime((input) -> input.toBuilder().completePartialNear(reference).build());
+        }
+
+        private SpaceWeatherAdvisoryImpl.Builder withCompleteNextAdvisory(final ZonedDateTime reference) {
+            requireNonNull(reference, "reference");
+            return mapNextAdvisory(nextAdvisory -> {
+                final NextAdvisoryImpl.Builder builder = NextAdvisoryImpl.Builder.from(nextAdvisory);
+                builder.mapTime(time -> time.toBuilder().completePartialNear(reference).build());
+                return builder.build();
+            });
+        }
+
+        private Builder withCompleteAnalysisTimes(final ZonedDateTime reference) {
+            requireNonNull(reference, "reference");
+            if (!getAnalyses().isEmpty()) {
```

Comment:
A minor note: using `mutateAnalyses()` might lead in slightly cleaner implementation, as you could directly modify the list when required.